### PR TITLE
Add experimental CV-CUDA resize

### PR DESCRIPTION
**Pull request by banasraf**
## Category:

**New feature**

## Description:

This PR adds new resize operator that uses CV-CUDA HQResize as its implementation.
It uses existing infrastructure of the resize operators to handle arguments and the cv-cuda op more or less replaces DALI kernel.

The testing involves extending existing tests to run the new operator in the same way the CPU and GPU versions are run.

### Affected modules and functionalities:

New operator, small changes in ResizeBase

### Tests:

- [ ] Existing tests apply
- [x] New tests added
  - [x] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [x] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [ ] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by banasraf** [comment]
!build

**Comment by banasraf** [comment]
!build

**Comment by banasraf** [comment]
!build

**Comment by banasraf** [comment]
!build

**Comment by banasraf** [comment]
!build

**Comment by banasraf** [comment]
!build

**Comment by banasraf** [comment]
!build

File: dali/operators/image/resize/experimental/resize_op_impl_cvcuda.h [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+    // Now that we have per-frame parameters, we can calculate the output shape of the
+    // effective frames (from videos, channel planes, etc).
+    GetResizedShape(out_shape_, in_shape_, make_cspan(params_), 0);
+
+    curr_minibatch_size_ = minibatch_size_;
+    // CV-CUDA tensors can't represent empty samples,
+    // so curr_minibatch_size_ is set to 1 to skip each empty sample one by one
+    if (HasEmptySamples(in_shape_)) {
+      curr_minibatch_size_ = 1;
+    }
```
Comment (mzient) [inline dali/operators/image/resize/experimental/resize_op_impl_cvcuda.h:56]: Why? Can't we just skip them when assembling the input TensorBatch?

**Comment by banasraf** [comment]
!build

File: dali/operators/image/resize/experimental/resize_op_impl_cvcuda.h (hunk 1 above)
Comment (banasraf) [reply dali/operators/image/resize/experimental/resize_op_impl_cvcuda.h]: Empty samples in the input are a very much edge case, I assume, so it wouldn't affect regular execution. Nevertheless, I implemented the solution with global index array

**Review by awolant — APPROVED** [review]
Looks like it works and is tested

**Review by szalpal — APPROVED** [review]
I believe it would be good to add explicit (and extensive) info, how does `fn.experimental.resize` differ from `fn.resize`. I know user can read CV-CUDA docs and then `fn.resize` docs, but putting the difference in `fn.experimental.resize` I believe is highly accurate. Plus, I don't think there's anywhere good explanation, how does DALI resize differ from HQResize.

**Pull request merged**